src/main/java/seedu/todolist/logic/ParserUtil.java

```diff
@@ -85,15 +87,21 @@ public static int parsePriority(String priorityString) throws InvalidPriorityExc
      * @throws InvalidDateException If the string is not in a valid date time format,
      *                              or if the parsed date is before the current time.
      */
-    public static LocalDateTime parseDeadline(String deadline) throws InvalidDateException {
+    public static LocalDateTime parseDeadline(String deadline) throws InvalidDateException, PassedDateException {
         if (deadline == null) {
             return null;
         }
 
         try {
             DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern(Formats.TIME_IN_1.getFormat()
                             + Formats.TIME_IN_2.getFormat()).withResolverStyle(ResolverStyle.STRICT);
-            return LocalDateTime.parse(deadline, inputFormatter);
+            // check with Shanghai time - same timezone as SG
+            if (!LocalDateTime.parse(deadline, inputFormatter).isAfter(LocalDateTime.now(ZoneId.of("CTT")))) {
```

Comment:
Time zone is not required - `LocalDateTime.now()` is already based on current time zone; this also limits the users to being in UTC+8, which can be considered a bug for users in other time zones